jwst/resample/resample.py

```diff
@@ -141,9 +143,27 @@ def __init__(self, input_models, output=None, single=False, blendheaders=True,
         self.pscale = pscale  # in deg
 
         log.debug('Output mosaic size: {}'.format(self.output_wcs.array_shape))
-        can_allocate, required_memory = datamodels.util.check_memory_allocation(
-            self.output_wcs.array_shape, kwargs['allowed_memory'], datamodels.ImageModel
-        )
+
+        allowed_memory = kwargs['allowed_memory']
+        if allowed_memory is None:
+            allowed_memory = os.environ.get('DMODEL_ALLOWED_MEMORY', allowed_memory)
```

Comment:
Note that this is the only use of this variable in the code.

There is one test that uses it:
https://github.com/spacetelescope/jwst/blob/e1215951096c2057df6abff6087b7f12d429ab25/jwst/regtest/test_fgs_guider.py#L46

Perhaps this should be deprecated/removed and it's usage replaced? Is there a way to provide this parameter to `calwebb_image3` without using an environment variable?

If it is kept, renaming it might be considered.